src/main/java/ch/njol/skript/effects/EffDestroyableKeys.java

```diff
@@ -0,0 +1,85 @@
+package ch.njol.skript.effects;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.aliases.ItemType;
+import ch.njol.skript.lang.Effect;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+import com.destroystokyo.paper.Namespaced;
+import org.bukkit.NamespacedKey;
+import org.bukkit.event.Event;
+import org.bukkit.inventory.meta.ItemMeta;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class EffDestroyableKeys extends Effect {
+
+	static {
+		Skript.registerEffect(EffDestroyableKeys.class,
+			"allow %~itemtypes% to (destroy|break|mine) %itemtypes/namespacedkeys%",
+			"prevent %~itemtypes% from (destroy|break|mine)[ing] %itemtypes/namespacedkeys%");
+		// could these patterns be combined into 1 using a parse tag? like `[:dis]allow` or smth?
+	}
+
+	@SuppressWarnings("NotNullFieldNotInitialized")
+	private Expression<ItemType> items;
+	@SuppressWarnings("NotNullFieldNotInitialized")
+	private Expression<?> modifyWith;
+	private boolean remove;
+
+	@Override
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		this.items = (Expression<ItemType>) exprs[0];
+		this.modifyWith = exprs[1];
+		this.remove = matchedPattern != 0;
+		return true;
+	}
+
+	@Override
+	protected void execute(Event event) {
+		ItemType[] items = this.items.getArray(event);
+		if (items.length == 0) {
+			return;
+		}
+
+		Set<Namespaced> keys = new HashSet<>();
+
+		for (Object o : modifyWith.getArray(event)) {
+			if (o instanceof ItemType item) {
+				keys.add(item.getMaterial().getKey());
```

Comment:
```suggestion
				keys.add(item.getRandom().getType().getKey());
```
`getMaterial()` only gets the first item of the ItemType, meaning it would always get `wooden sword` from `any sword`.
Speaking of which, we may want to have this handle all the itemdatas in an itemtype, so that you can do
```
prevent x from breaking any log
```

Reply:
How would that work? I'm not super familiar with how the `any` and `all` aliases work. Would it just involve looping through some method?

Reply:
Yeah, see ItemData for the "true" representation of the items within an itemtype. It's pretty awkward and likely subject to change in a future rework, but it's what we have for now.

Reply:
`prevent x from breaking any log`, correct me if I'm wrong but we would need to also do `ItemType#setAll()` here otherwise when we use `getAll()` it'll return a single element from `any log`